src/components/editMeasure/testCases/components/testCaseLanding/common/shiftDates/ShiftDatesDialog.tsx

```diff
@@ -34,7 +47,72 @@ const ShiftDatesDialog = ({
   });
 
   const handleSubmit = async (value) => {
-    onTestCaseShiftDates(testCases, value.shiftDatesInput);
+    if (isQdm) {
+      testCaseService.current
+        .shiftQdmTestCaseDates(
+          measure.id,
+          testCases.map((testCase) => testCase.id),
+          value.shiftDatesInput
+        )
+        .then((response) => {
+          if (response.length === 0) {
+            setToastOpen(true);
+            setToastType("success");
+            setToastMessage(`All Test Case dates successfully shifted.`);
+          } else {
+            setWarnings((prevState) => [...prevState, ...response]);
+          }
+        })
+        .catch((err) => {
+          setToastOpen(true);
+          setToastType("danger");
+          if (
+            featureFlags?.Locking &&
+            err?.Error?.includes("locked by another user")
+          ) {
+            setToastMessage(
+              `One or more of the Test Cases are locked by another user. Test Case Dates cannot be shifted.`
```

Comment:
her as well, I dont think we need to override the messages when same message is returned by service. if required we can update them in service itself. and just use `err.message` here